google-cloud-storage/src/test/java/com/google/cloud/storage/BucketInfoTest.java

```diff
@@ -333,4 +337,22 @@ public void testLogging() {
     assertEquals("test-bucket", logging.getLogBucket());
     assertEquals("test-", logging.getLogObjectPrefix());
   }
+
+  @Test
+  public void testDeleteLifecycleRules() {
+    BucketInfo bucketInfo = EasyMock.createStrictMock(BucketInfo.class);
+    expect(bucketInfo.deleteLifecycleRules()).andReturn(ImmutableList.of(LIFECYCLE_RULES.get(0)));
+    replay(bucketInfo);
+    List<LifecycleRule> actualResults = bucketInfo.deleteLifecycleRules();
+    assertThat(actualResults).hasSize(1);
+  }
+
+  @Test
+  public void testDeleteLifecycleRuleIfNotExists() {
+    BucketInfo bucketInfo = EasyMock.createStrictMock(BucketInfo.class);
+    expect(bucketInfo.deleteLifecycleRules()).andReturn(ImmutableList.<LifecycleRule>of());
+    replay(bucketInfo);
+    List<LifecycleRule> lifecycleRules = bucketInfo.deleteLifecycleRules();
+    assertThat(lifecycleRules).isEmpty();
+  }
 }
```

Comment:
The original idea of `testDeleteLifecycleRuleIfNotExists()` was to cover the case when the user tries to delete a rule which doesn't exist. Now, the method under test has no parameters, so this test is redundant. 
To make this test useful, you can verify that `deleteLifecycleRules` does not cause an extra RPC call if nothing to delete.